src/utils/cosmosdb_model.ts

```diff
@@ -243,12 +244,17 @@ export abstract class CosmosdbModel<
     query: string | SqlQuerySpec,
     options?: FeedOptions
   ): TaskEither<CosmosErrors, Option<TR>> {
+    this.container.items.query.bind(this.container.items);
+    // tslint:disable-next-line: no-dead-store
     const queryIterator = this.container.items.query<TR>(query, options)
       .fetchAll;
     return tryCatch<
       CosmosErrors,
       PromiseType<ReturnType<typeof queryIterator>>
-    >(() => queryIterator(), toCosmosErrorResponse)
+    >(
+      () => this.container.items.query<TR>(query, options).fetchAll(),
+      toCosmosErrorResponse
+    )
```

Comment:
@cloudify: Can you check this? With these changes integration tests run correctly, but unit test fails and viceversa.

Reply:
ok let me refactor this a bit